src/stores/appStore.ts

```diff
@@ -9,101 +8,63 @@ import {
         async initialiseData() {
-            this.appConfig = await getAppConfig();
-
-            // Read all adm1 indicators from a single file
-            const allIndicators = await getGlobalIndicators(1);
-            // Load all admin1 geojson - load simplified boundaries only, from a
-            // single file,
-            const allGeojson = await getGlobalGeojsonFeatures(1);
+            const state = this as AppState;
+            state.appConfig = await getAppConfig();
+            state.countryBoundingBoxes = await getCountryBoundingBoxes();
+            state.admin1Geojson = await getGlobalGeojsonFeatures(1);
+            state.admin1Indicators = await getGlobalIndicators(1);
 
-            this.countryBoundingBoxes = await getCountryBoundingBoxes();
-            this.admin1Geojson = allGeojson;
-            this.admin1Indicators = allIndicators;
+            state.initialisationComplete = true;
         },
-        async selectCountry(countryId: string) {
-            if (countryId === this.selectedCountryId) {
-                return;
-            }
+        async updateMapSettings(newMapSettings: MapSettings) {
+            const state = this as AppState;
+            const { country, adminLevel } = newMapSettings;
 
-            if (!countryId) {
-                this.selectedCountryId = "";
-                this.admin0GeojsonFeature = null;
+            if (!country) {
+                state.mapSettings = newMapSettings;
+                state.admin0GeojsonFeature = null;
                 return;
             }
 
+            let level = adminLevel;
             // Some countries do not have admin2 regions or data - if one of these is selected, we load
             // a more detailed geojson, and re-use its admin1 indicators as "admin2"
-            const level = this.appConfig.countriesWithoutAdmin2.includes(countryId) ? 1 : 2;
+            const admin2DataMissing = state.appConfig.countriesWithoutAdmin2.includes(country);
+            if (level === 2 && admin2DataMissing) {
+                level = 1;
+                state.admin2Indicators[country] = state.admin1Indicators[country];
+            }
 
-            if (!(countryId in this.admin2Indicators)) {
-                if (level === 1) {
-                    this.admin2Indicators[countryId] = this.admin1Indicators[countryId];
-                } else {
-                    this.admin2Indicators[countryId] = await getIndicators(countryId, level);
-                }
+            if (level === 2 && !admin2DataMissing && !(country in state.admin2Indicators)) {
+                state.admin2Indicators[country] = await getIndicators(country, level);
             }
-            if (!(countryId in this.admin2Geojson)) {
-                this.admin2Geojson[countryId] = await getGeojsonFeatures(countryId, level);
+
+            if (!(country in state.admin2Geojson)) {
+                state.admin2Geojson[country] = await getGeojsonFeatures(country, level);
             }
```

Comment:
```suggestion
state.admin2Geojson[country] = state.admin2Geojson[country] || await getGeojsonFeatures(country, level);
```

This achieves the same thing and is terser, but maybe it's less readable? (If people like the terse option, it can apply a few lines back, too.)

Reply:
im happy to do that, i think it is readable but lets confirm with @EmmaLRussell

Reply:
Actually, terser still is using OR assignment:

```ts
state.admin2Geojson[country] ||= await getGeojsonFeatures(country, level);
```

Reply:
I'd vote for the original suggestion personally, not a huge fan of OR assignment, though it's really just aesthetic preference at this point.

Reply:
can do the original suggestion then! done